chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt

```diff
@@ -233,45 +348,90 @@ class CurrentUser(
+     * out the [docs][https://pusher.com/docs/chatkit/reference/android].
+     * The [callback] will be called with a [Result] when the operation is complete. If the
+     * operation was successful you will receive an [Int] which contains the message id, or an
+     * [Error] informing you of what went wrong.
+     */
     fun sendMultipartMessage(
             room: Room,
             parts: List<NewPart>,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMultipartMessage(room, parts) }, callback)
 
+    /**
+     * @see [sendMultipartMessage]
+     */
     fun sendMultipartMessage(
             roomId: String,
             parts: List<NewPart>,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMultipartMessage(roomId, parts) }, callback)
 
+    /**
+     * Sets that the current user is typing in the [room]. You can send as many of these as you like,
+     * the SDK will handle rate limiting the requests. The [callback] will be called with a [Result]
+     * when the operation is complete. If the operation was successful you will receive a [Unit], or
+     * an [Error] informing you of what went wrong.
+     */
     fun isTypingIn(room: Room, callback: (Result<Unit, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.isTypingIn(room) }, callback)
 
+    /**
+     * @see [isTypingIn]
+     */
     fun isTypingIn(roomId: String, callback: (Result<Unit, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.isTypingIn(roomId) }, callback)
 
+    /**
+     * Get a list of the public rooms that you are not a member of yet. The [callback] will be called
+     * with a [Result] when the operation is complete. If the operation was successful you will receive
+     * a List<Room>, or an [Error] informing you of what went wrong.
+     */
     fun getJoinableRooms(callback: (Result<List<Room>, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.getJoinableRooms() }, callback)
 
+    /**
+     * Get the list of users in the [room]. The [callback] will be called with a [Result] when the
+     * operation is complete. If the operation was successful you will receive a List<User>, or an
+     * [Error] informing you of what went wrong.
+     */
     fun usersForRoom(room: Room, callback: (Result<List<User>, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.usersForRoom(room) }, callback)
 
+    /**
+     * @see [usersForRoom]
+     */
     fun usersForRoom(roomId: String, callback: (Result<List<User>, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.usersForRoom(roomId) }, callback)
 
+    /**
+     * Enables push notifications for the current user. The [callback] will be called with a [Result]
+     * when the operation is complete. If the operation was successful you will receive a [Unit],
+     * or an [Error] informing you of what went wrong.
```

Comment:
We should probably link to the disable method on `ChatManager` and mention in what circumstances it should be called (e.g. user log out).

We could be more specific about what it means to enable push notifications here as well (register this device against this user), and how this method is intended to be used e.g. it's safe and reasonable to call it every time the SDK is initialised, it will de-register the previous userId if a different registration existed for the device, etc.

Reply:
Definitely we should tell people how to sign out.

I think we should tell people it's fine to call it every time the user connects successfully, and that the current user can be registered to up to 10 android devices at any given time. However, we should also note that they should handle disabling notifications whenever they detect the user is in the logged out mode as per the comment in https://pusher.com/docs/chatkit/guides/push_notifications/setup-android#disabling-push-notifications